File: app/services/history_service.py
from app import db
from app.models import History

class HistoryService:
    @staticmethod
    def create_history(user_id, problem_num, solve_time):
        new_history = History(
            user_id=user_id,
            problem_num=problem_num,
            solve_time=solve_time
        )

        db.session.add(new_history)
        db.session.commit()
        return new_history

File: server/src/main/kotlin/com/reminder/server/domain/history/HistoryService.kt
package com.reminder.server.domain.history

import com.reminder.server.domain.user.UserRepository
import com.reminder.server.global.exception.UserNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class HistoryService(
    private val historyRepository: HistoryRepository,
    private val userRepository: UserRepository,
) {
    @Transactional
    fun saveHistory(userId: Long, problemNum: String, solveTime: String): History {
        val user = userRepository.findById(userId).orElseThrow { UserNotFoundException(userId) }
        return historyRepository.save(History(user, problemNum, solveTime))
    }
}
